### ISSUE TITLE
[$250] Improvement: Make the "press the green + button" text in the "Submit an Expense" task on mobile a clickable emoji

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#54643]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.79-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation** (hyperlinked to channel name): [expensify_bugs](https://expensify.slack.com/archives/C049HHMV9SM/p1735435313879969)

## Action Performed:
1. Login with new account
2. Choose "Get paid back by my employer" option in onboarding
3. Click the "Submit expense" task from Concierge

## Expected Result:
Green + button visible or hyperlinked in the task 

Create a :global_create: emoji in NewDot that copies :global-create:
Add a hyperlink like expensify://global-create that just opens Global Create
Update the text to be 1. Press the [:global-create: button](expensify://global-create)
This way:
The :global-create: will be visible right there
They can click it without leaving the task

## Actual Result:
Green + button is not visible and not hyperlinked in the task 

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

<img src="https://github.com/user-attachments/assets/afcd2c14-dcab-48c9-80d8-c81e61a53fe5" width="40%" height="40%">

<img src="https://github.com/user-attachments/assets/f619073b-90cc-414c-ba31-1525bc4ec280" width="40%" height="40%">

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021874860195609179779</li>
        <li>Upwork Job ID: 1874860195609179779</li>
        <li>Last Price Increase: 2025-01-09</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @joekaufmanexpensify</details>

**Comment by anmurali** [comment]
We need to actually create this FE element I think? It doesn't exist right now...

**Comment by anmurali** [comment]
Turns out we need to actually figure out how to support custom emojis or inline images to accomplish this. As a trivial to implement improvement, let's just make `Click the green + button` a hyperlink that opens the global create menu without leaving the task.

**Comment by Amoralchik** [comment]
## Proposal

According to https://github.com/Expensify/App/issues/54643#issuecomment-2568068089

### Please re-state the problem that we are trying to solve in this issue.

Make the "press the green + button" text in the "Submit an Expense" task clickable

### What is the root cause of that problem?

"press the green + button" is plain text by now

### What changes do you think we should make in order to solve the problem?

First, we need to add 
```js
createExpenseLink: `${environmentURL}/${ROUTES.MONEY_REQUEST_CREATE_TAB_MANUAL.getRoute(
    CONST.IOU.ACTION.CREATE,
    CONST.IOU.ACTION.CREATE,
    CONST.IOU.OPTIMISTIC_TRANSACTION_ID,
    onboardingPolicyID ?? '-1',
)}`,
```
to:
https://github.com/Expensify/App/blob/c6d8045d8c6923fd5bf54fef1f0c9f455fbc1d88/src/libs/actions/Report.ts#L3620-L3630

then we can use it [in](https://github.com/Expensify/App/blob/15d0dabe7825fc7a62720b4744788a3d911ad47c/src/CONST.ts#L154) to change description:
```js
description: ({createExpenseLink}) => {
    return (
        `*Submit an expense* by entering an amount or scanning a receipt.\n` +
        `\n` +
        `Here’s how to submit an expense:\n` +
        `\n` +
        `1. Click the [green *+* button](${createExpenseLink}) \n` +
        `2. Choose *Create expense*.\n` +
        `3. Enter an amount or scan a receipt.\n` +
        `4. Add your reimburser to the request.\n` +
        `5. Click *Submit*.\n` +
        `\n` +
        `And you’re done! Now wait for that sweet “Cha-ching!” when it’s complete.`
    );
},
```

**Comment by c3024** [comment]
The Slack [discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1736055810999609?thread_ts=1735435313.879969&cid=C049HHMV9SM) and this comment [here](https://github.com/Expensify/App/issues/54643#issuecomment-2568068089) need opening the menu containing the "Create expense" on clicking the step "Click the green + button" so that the second step of clicking on the "Create expense" can be done there itself without leaving the task report page. But, the proposals are linking the task step to open the RHP of create expense flow.

Can you please update the proposals to match the requirement? Thanks!

**Comment by anmurali** [comment]
I think we might actually take this back and research a clickable custom emoji after all.

**Comment by c3024** [comment]
Yes, I will work on it. Thanks!!

**Comment by Amoralchik** [comment]
Proposal Update

**Comment by c3024** [comment]
Thanks for the update, @Amoralchik.

Based on [this discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1736278669423269?thread_ts=1736060629.621839&cid=C01GTK53T8Q), I too think that using `react-native-render-html` to open the FAB menu is the better approach.

@Amoralchik’s [proposal](https://github.com/Expensify/App/issues/54643#issuecomment-2571421279) works well and looks good to me. Further refinements can be made during the PR process.

🎀 👀 🎀 C+ reviewed!

**Comment by Amoralchik** [comment]
PR will be ready Tomorrow, Thanks.

**Comment by Amoralchik** [comment]
[PR](https://github.com/Expensify/App/pull/55222) is finally ready for review

**Comment by Amoralchik** [comment]
Apologies for the delay. I’ve been working to address all the issues, but unfortunately, progress is limited at this time.

- **PR**: [GitHub PR #56602](https://github.com/Expensify/App/pull/56602)
  - Issue [56308](https://github.com/Expensify/App/issues/56308): Fixed by escaping `shouldEscapeText` for `['submitExpense', 'trackExpense', 'startChat', 'splitExpense']`.
  - Issue [56356](https://github.com/Expensify/App/issues/56356): Fixed by returning to the original logic using `useContext`. However, now I have an issue where you can close the `PopoverMenu` by clicking on an emoji.
  - Issue [56341](https://github.com/Expensify/App/issues/56341): Still unresolved. Despite incorporating good proposal from @roryabraham, I’m encountering strange issues with emoji alignment on both IOS and Android Native Apps.

I’m unsure how to proceed further at this point and may need additional support. All changes have been included in the PR for review.

**Comment by anmurali** [comment]
We had to revert due to issues and @Amoralchik is working on a new PR. @c3024 can you pls help https://github.com/Expensify/App/issues/54643#issuecomment-2647941833

**Comment by c3024** [comment]
>Issue https://github.com/Expensify/App/issues/56356: Fixed by returning to the original logic using useContext. However, now I have an issue where you can close the PopoverMenu by clicking on an emoji.

This happens because clicking outside the FAB toggles it, and simultaneously, clicking on the emoji toggles it again. This fix is specifically for narrow screens, so I think this behavior is fine.

>Issue https://github.com/Expensify/App/issues/56341: Still unresolved. Despite incorporating good proposal from @roryabraham, I’m encountering strange issues with emoji alignment on both IOS and Android Native Apps.

I’ve been trying to find a fix for this but haven’t found a decisive solution yet.

**Comment by roryabraham** [comment]
> Still unresolved. Despite incorporating good proposal from @roryabraham, I’m encountering strange issues with emoji alignment on both IOS and Android Native Apps.

Hmmm - what seems to be the problem? Can you share a screenshot perhaps?

**Comment by c3024** [comment]
I have not found a way to position the emoji correctly for all different font sizes on Android. Still exploring.

**Comment by roryabraham** [comment]
@c3024 I appreciate that you're looking into this, but you should start a slack thread and leave notes on your explanation in a public place. That way we can follow along and also tag people who might be able

**Comment by c3024** [comment]
>Issue #56341 is still unresolved. Despite incorporating a good proposal from @roryabraham, I’m encountering strange issues with emoji alignment on both iOS and Android native apps.

We cap the maximum font size here:

https://github.com/Expensify/App/blob/d7ecbe3a8d4b09226e683b189bad4010de5168a2/src/styles/variables.ts#L10-L12

If we simply multiply image sizes with `fontScale`, the font size is restricted to the max value for large scales, but the image size increases proportionally with `fontScale`, becoming too large and breaking alignment.

To fix this, we should use one of the values that rely on `getValueUsingPixelRatio` for image size as well. Since we use `fontSizeNormal` for the description text, I think we should use the same value for the image width and height too, with a slight margin adjustment. This fixes the issue on iOS.

On Android, scaling affects the alignment of inline images in text. I tried various approaches and found that a slightly greater margin adjustment based on `fontScale` is needed for correct emoji rendering.

I’ve added these working code change suggestions in the draft PR.

@Amoralchik @roryabraham Please have a look!

I’ll bring this discussion to Slack if we prefer a better solution.

**Comment by c3024** [comment]
> > Still unresolved. Despite incorporating good proposal from [@roryabraham](https://github.com/roryabraham), I’m encountering strange issues with emoji alignment on both IOS and Android Native Apps.
> 
> Hmmm - what seems to be the problem? Can you share a screenshot perhaps?
> 
> I wonder if the problem is due to one of these 🐛 in RN core:
> 
> * this bug that prevents hot-reloading with changes to fontScale: [No hot reloading when change font scale facebook/react-native#45655](https://github.com/facebook/react-native/issues/45655)
> * [[iOS] increasing font scale (content size catgory) breaks text facebook/react-native#45857](https://github.com/facebook/react-native/issues/45857) which results in text being cut off on iOS

Yea, there is a problem with hot reloading but we had issues with alignment too. iOS works reliably if we use a value based on `getValueUsingPixelRatio`. So, these two RN issues do not seem to be related to our present issue.

**Comment by roryabraham** [comment]
Slack discussion: https://expensify.slack.com/archives/C01GTK53T8Q/p1739919755695589

**Comment by roryabraham** [comment]
@c3024 sounds like we're ready to move forward with the simple approach you suggested 👍🏼

**Comment by c3024** [comment]
Not overdue @MelvinBot . @Amoralchik is working on the PR!

**Comment by rushatgabhane** [comment]
hey just a thought - we should unit tests for fab button if we change the way it works. it will help prevent [regressions](https://github.com/Expensify/App/issues/57797)

**Comment by Amoralchik** [comment]
In addition to @rushatgabhane 

Fix PR here: https://github.com/Expensify/App/pull/57874
and I don't think we need to resolve issues with button lags in this PR, according to my comment: https://github.com/Expensify/App/pull/57874#issuecomment-2702294017

**Comment by rushatgabhane** [comment]
im sorry, that PR makes the FAB button really laggy. if it is an underlying issue, we should fix that first because we cannot compromise on quality of the green FAB button

**Comment by Amoralchik** [comment]
@rushatgabhane as I say, that's not PR issues, you can reproduce this lags it in main right now

**Comment by rushatgabhane** [comment]
latest `main` does not lag for me. maybe im missing something?

https://github.com/user-attachments/assets/8c2ac06d-de1a-4156-849a-7d9ae2ff6248

**Comment by Amoralchik** [comment]
@rushatgabhane i don't know, I got this on latest (i made git pull right now)

https://github.com/user-attachments/assets/d66dad8c-c0fa-4483-9505-ce87fcbc43ff

**Comment by joekaufmanexpensify** [comment]
@Amoralchik @rushatgabhane @c3024 I am just catching up here. Is the consensus that the PR here did introduce some regressions, or no?

**Comment by rushatgabhane** [comment]
yes we reverted the pr here https://github.com/Expensify/App/pull/57890

**Comment by joekaufmanexpensify** [comment]
Sounds good. @Amoralchik is there an ETA for the updated PR to fix the issues?

**Comment by Amoralchik** [comment]
@rushatgabhane So, can you say, you can reproduce that https://github.com/Expensify/App/issues/54643#issuecomment-2705193940 on the latest main? I see you placed emoji, but I don't think I understand it correctly

@joekaufmanexpensify you can find PR here: https://github.com/Expensify/App/pull/57874 but for now we have a little strange behavior with FAB, that in my version happens in both main and my branch (https://github.com/Expensify/App/issues/54643#issuecomment-2705193940 so I don't think is issue that we need to fix in my PR) and according to @rushatgabhane occurs only in my branch (https://github.com/Expensify/App/issues/54643#issuecomment-2705186564)

**Comment by joekaufmanexpensify** [comment]
Discussing [here. ](https://expensify.slack.com/archives/C07HPDRELLD/p1741616201820909?thread_ts=1741063217.759549&cid=C07HPDRELLD) @c3024 is going to jump in and review the new PR.

**Comment by joekaufmanexpensify** [comment]
@c3024 could you handle the checklist so we can prep for payment on Monday?

**Comment by c3024** [comment]
@joekaufmanexpensify, the PR was reverted. New PR #57874 in works.

**Comment by joekaufmanexpensify** [comment]
Ah, I didn't realize the automation was for the PR that got reverted. TY for confirming! I removed the note that this is waiting for payment now.

**Comment by joekaufmanexpensify** [comment]
PR is pending review from @c3024. Can you prioritize that this week?

**Comment by quinthar** [comment]
Sounds like we are ditching the previous PR and moving to this: https://github.com/Expensify/App/pull/59071 -- @c3024 gave an ETA of tomorrow [here](https://expensify.slack.com/archives/C07HPDRELLD/p1743095241850659?thread_ts=1742275825.663609&cid=C07HPDRELLD).

**Comment by joekaufmanexpensify** [comment]
Ongoing discussion on new PR [here](https://expensify.slack.com/archives/C07HPDRELLD/p1743442439947689?thread_ts=1742275825.663609&cid=C07HPDRELLD).

**Comment by joekaufmanexpensify** [comment]
Variety of PRs for the new solution are in review.

**Comment by joekaufmanexpensify** [comment]
One PR merged last week, still several in review.

**Comment by quinthar** [comment]
What's the final design we went for, did we:
1. Add `:global-create:` to our custom font, or
2. Add `:global-create:` to a new font

**Comment by shawnborton** [comment]
2 - this way we don't have to re-export our actual font files each time a new emoji is made, now we'll have an additional font that is only for adding custom emojis. PR is [here](https://github.com/Expensify/App/pull/59301) if you want to take it for a spin, try the test builds [here](https://github.com/Expensify/App/pull/59301#issuecomment-2791928128).

**Comment by joekaufmanexpensify** [comment]
Update [here ](https://expensify.slack.com/archives/C07HPDRELLD/p1745206931640189?thread_ts=1745204525.350809&cid=C07HPDRELLD). Next step is to get https://github.com/Expensify/react-native-live-markdown/pull/661 merged asap to unblock https://github.com/Expensify/App/pull/59301

**Comment by Amoralchik** [comment]
@danielrvidal, I'm not entirely sure. From what I can see, my changes related to the clickable logic were not merged and were eventually closed. It looks like @c3024 added the custom emoji, but without the clickable functionality.

**Comment by joekaufmanexpensify** [comment]
Discussing more [here](https://expensify.slack.com/archives/C07HPDRELLD/p1753127629885859?thread_ts=1752555249.748139&cid=C07HPDRELLD)

**Comment by joekaufmanexpensify** [comment]
We landed on calling this one finished. [Discussing payment](https://expensify.slack.com/archives/C01SKUP7QR0/p1753196271915169).

**Comment by joekaufmanexpensify** [comment]
We [discussed in slack](https://expensify.slack.com/archives/C01SKUP7QR0/p1753196271915169) and landed on the following for payment:

- @Amoralchik $250 via upwork for their PRs that attempted to create the new custom emoji and make it clickable. 
- @c3024 $500 via NewDot for their PRs that created the new custom emoji and for reviewing @Amoralchik's PRs. 
- @eVoloshchak $250 via NewDot for reviewing @c3024's App PR that created the new custom emoji.

**Comment by joekaufmanexpensify** [comment]
OG upwork job is closed, so I opened a [new one ](https://www.upwork.com/jobs/~021948038172164934401)
